### ISSUE TITLE
Base_value not matching

### ISSUE THREAD

**Issue by nirupamkar** [shap/shap#475]
I am trying to match the base value with the force plot. I believe the base value is the weighted average of prediction in training data. The below example is for a weighted regression problem with lightgbm 

`import numpy as np
import pandas as pd
import lightgbm as lgb
categorical_features =  ['c1','c2'] 
np.random.seed(0)
matrix = np.random.choice(['a','b','c','d'],100*2)
matrix.resize(100,2)

df = pd.DataFrame(matrix,columns = categorical_features)
df['target'] = np.where(df.c1=="a",np.random.normal(1,0.2),np.where(df.c2=="b",np.random.normal(2,0.5),np.random.normal(3,0.5)))

def as_cat(df):
    for feature_name in categorical_features:
        df[feature_name] = df[feature_name].astype('category')
    return df

def model_train(X_train):
    X_train = as_cat(X_train) 
    valid_features=categorical_features
        
    parameters={
                'objective': 'regression', "num_leaves":3,
                 'boosting_type': 'gbdt','leanring_rate':.1}

    lgb_train_data = lgb.Dataset(X_train[valid_features], label=X_train.target, 
                                categorical_feature=categorical_features,weight=X_train["weight"].values,
                                          free_raw_data=False)
    print "running "

    model = lgb.train(parameters,
        lgb_train_data,verbose_eval=0,
        num_boost_round=1,valid_sets=[lgb_train_data], 
                             valid_names=['train'] )

    return model

df["weight"]=np.random.normal(2,3,df.shape[0])
df = as_cat(df)
model = model_train(df)

explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(df[categorical_features])
shap.force_plot(explainer.expected_value, shap_values[2,:], df[categorical_features].iloc[2,:])`

In my case
Weighted_average=2.8460
Simple_average=2.60427
base_value=2.818

Could you please help me to understand this or I am missing something? FYI, it's matching for unweighted cases

**Comment by nirupamkar** [comment]
Well, after plotting the tree, I think the base value is the weighted mean of leaf node prediction and weight being the leaf_count. 
So during lightgbm model building "leaf node prediction " has been done using pre defined weighing scheme. However while calculating base_value the weighing is just the count. Is this correct?

**Comment by slundberg** [comment]
That's right. TreeExplainer currently does not consider any of the weighting that was in the training dataset. This is because the node coverage values recorded by LightGBM in the tree do not reflect any sample weighting. I am adding a to-do tag, since being able to handle weighting seems like a good feature to have. Note that right now you could redefine the `explainer.model.node_sample_weight` matrix yourself and then set `self.model.model_type` to `"internal"`. That would allow you to run TreeExplainer on a model that you got to pick the weighting scheme for.

**Comment by nirupamkar** [comment]
@slundberg  thanks for the reply. Could you please help me which code I need to modify?
